evalml/automl/callbacks.py

```diff
@@ -0,0 +1,47 @@
+from evalml.exceptions import PipelineScoreError
+from evalml.utils.logger import get_logger
+
+logger = get_logger(__file__)
+
+
+def silent_error_callback(exception, automl, **kwargs):
+    """No-op."""
+    pass
+
+
+def raise_error_callback(exception, automl, **kwargs):
+    """Raises the exception thrown by the AutoMLSearch object. Also logs the exception as an error."""
+    logger.error(f'AutoMLSearch raised a fatal exception: {str(exception)}')
+    raise exception
+
+
+def log_and_save_error_callback(exception, automl, **kwargs):
+    """Logs the exception thrown by the AutoMLSearch object as a warning
+        and adds the exception to the 'errors' list in AutoMLSearch object results."""
+    logger.warning(f'AutoML search encountered an exception: {str(exception)}')
+    automl._results['errors'] = automl._results.get('errors', [])
```

Comment:
Ah, slick.

If I had to choose between this and adding code to `AutoMLSearch.__init__` which initializes "errors" key in automl results to an empty list, I think I'd choose the latter, simply because that way there's *always* an "errors" key in the results, even if its empty, and that feels like simpler behavior to me.